packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts

```diff
@@ -1,125 +1,203 @@
-    const { data, layout } = jsonObj;
-    const mapXToDataPoints: { [key: string]: IVerticalStackedChartProps } = {};
-    let yMaxValue = 0;
-
-    data.forEach((series: any, index1: number) => {
-      series.x.forEach((x: string | number, index2: number) => {
-        if (!mapXToDataPoints[x]) {
-          mapXToDataPoints[x] = { xAxisPoint: x, chartData: [], lineData: [] };
-        }
-        if (series.type === 'bar') {
-          mapXToDataPoints[x].chartData.push({
-            legend: series.name,
-            data: series.y[index2],
-            color: series.marker?.color || getNextColor(index1),
-          });
-        } else if (series.type === 'line') {
-          mapXToDataPoints[x].lineData!.push({
-            legend: series.name,
-            y: series.y[index2],
-            color: series.marker?.color || getNextColor(index1),
-          });
-        }
-        yMaxValue = Math.max(yMaxValue, series.y[index2]);
-      });
+  const { data, layout } = jsonObj;
+  const mapXToDataPoints: { [key: string]: IVerticalStackedChartProps } = {};
+  let yMaxValue = 0;
+
+  data.forEach((series: any, index1: number) => {
+    series.x.forEach((x: string | number, index2: number) => {
+      if (!mapXToDataPoints[x]) {
+        mapXToDataPoints[x] = { xAxisPoint: x, chartData: [], lineData: [] };
+      }
+      if (series.type === 'bar') {
+        mapXToDataPoints[x].chartData.push({
+          legend: series.name,
+          data: series.y[index2],
+          color: series.marker?.color || getNextColor(index1),
+        });
+      } else if (series.type === 'line') {
+        mapXToDataPoints[x].lineData!.push({
+          legend: series.name,
+          y: series.y[index2],
+          color: series.marker?.color || getNextColor(index1),
+        });
+      }
+      yMaxValue = Math.max(yMaxValue, series.y[index2]);
     });
-  
+  });
+
+  return {
+    data: Object.values(mapXToDataPoints),
+    chartTitle: layout.title,
+    // width: layout.width,
+    // height: layout.height,
+    barWidth: 'auto',
+    yMaxValue,
+  };
+};
+
+export const transformPlotlyJsonToLineChartProps = (jsonObj: any): ILineChartProps => {
+  const { data, layout } = jsonObj;
+
+  const chartData: ILineChartPoints[] = data.map((series: any, index: number) => {
     return {
-      data: Object.values(mapXToDataPoints),
-      chartTitle: layout.title,
-      // width: layout.width,
-      // height: layout.height,
-      barWidth: 'auto',
-      yMaxValue,
+      legend: series.name,
+      data: series.x.map((x: string | number, i: number) => ({
+        x: new Date(x),
+        y: series.y[i],
+      })),
+      color: series.line.color || getNextColor(index),
     };
+  });
+
+  const chartProps: IChartProps = {
+    chartTitle: layout.title || '',
+    lineChartData: chartData,
+  };
+
+  return {
+    data: chartProps,
+  };
+};
+
+export const transformPlotlyJsonToAreaChartProps = (jsonObj: any): IAreaChartProps => {
```

Comment:
looks like line and area chart transformation logic is same. Use same function for both.

Reply:
merged them